lib/pbench/test/unit/server/database/test_index_map_db.py

```diff
@@ -46,49 +34,26 @@ def test_merge(self, db_session, attach_dataset, m1, m2):
 
         maps = [
             {
-                "run-data": {"prefix.run-data.2023-07": ["id1", "id2"]},
-                "run-toc": {
-                    "prefix.run-toc.2023-06": ["id3", "id4"],
-                    "prefix.run-toc.2023-07": ["id5", "id6"],
-                },
-                "funky-data": {"prefix.funky-data.1918-02": ["id1", "id2"]},
+                "run-data": ["prefix.run-data.2023-07"],
+                "run-toc": ["prefix.run-toc.2023-06", "prefix.run-toc.2023-07"],
+                "funky-data": ["prefix.funky-data.1918-02"],
             },
             {
-                "run-data": {"prefix.run-data.2023-06": ["id8", "id9"]},
-                "run-toc": {
-                    "prefix.run-toc.2023-06": ["id10", "id12", "id13", "id14"],
-                    "prefix.run-toc.2023-08": ["id20", "id21"],
-                },
-                "run-sample": {"prefix.run-sample.2023-08": ["id22", "id23"]},
+                "run-data": ["prefix.run-data.2023-06"],
+                "run-toc": ["prefix.run-toc.2023-06", "prefix.run-toc.2023-08"],
+                "run-sample": ["prefix.run-sample.2023-08"],
             },
         ]
         dataset = Dataset.query(name="drb")
         assert not IndexMap.exists(dataset)
-        assert list(IndexMap.stream(dataset)) == []
 
         IndexMap.create(dataset, maps[m1])
         IndexMap.merge(dataset, maps[m2])
-
-        assert sorted(IndexMap.stream(dataset), key=lambda i: (i.id, i.index)) == [
-            IndexStream("prefix.funky-data.1918-02", "id1"),
-            IndexStream("prefix.run-data.2023-07", "id1"),
-            IndexStream("prefix.run-toc.2023-06", "id10"),
-            IndexStream("prefix.run-toc.2023-06", "id12"),
-            IndexStream("prefix.run-toc.2023-06", "id13"),
-            IndexStream("prefix.run-toc.2023-06", "id14"),
-            IndexStream("prefix.funky-data.1918-02", "id2"),
-            IndexStream("prefix.run-data.2023-07", "id2"),
-            IndexStream("prefix.run-toc.2023-08", "id20"),
-            IndexStream("prefix.run-toc.2023-08", "id21"),
-            IndexStream("prefix.run-sample.2023-08", "id22"),
-            IndexStream("prefix.run-sample.2023-08", "id23"),
-            IndexStream("prefix.run-toc.2023-06", "id3"),
-            IndexStream("prefix.run-toc.2023-06", "id4"),
-            IndexStream("prefix.run-toc.2023-07", "id5"),
-            IndexStream("prefix.run-toc.2023-07", "id6"),
-            IndexStream("prefix.run-data.2023-06", "id8"),
-            IndexStream("prefix.run-data.2023-06", "id9"),
-        ]
+        indices = set()
+        for m in maps:
+            for i in m.values():
+                indices.update(i)
```

Comment:
Yeah, at the time I thought a 2-level comprehension would be significantly less clear. I think that now, looking at both, it's perhaps less clear that it's less clear, but I'm not inclined to change it right now since I don't think it represents a problem.

Reply:
> I thought a 2-level comprehension would be significantly less clear.

My comprehension is actually _3-level_.  😏  Your use of `List.update()` encapsulates the third level.